internal/bminventory/inventory.go

```diff
@@ -198,6 +199,7 @@ type bareMetalInventory struct {
 	gcConfig             garbagecollector.Config
 	providerRegistry     registry.ProviderRegistry
 	insecureIPXEURLs     bool
+	httpListenPortStr    string
```

Comment:
Other artifacts access the image service.

Reply:
Not the ipxe script which we are currently exposing conditionally as http from the assisted service [here](https://github.com/openshift/assisted-service/blob/6f2f295b170c39550e59e30b1d49356d128f5a67/internal/controller/controllers/infraenv_controller.go#L533-L555)

Currently it looks like we're relying on this working with a route which would not require a port and would assume the defaults for http(s). This works in the route case, but wouldn't if we were exposing http and https simultaneously and also not using an openshift route.

Fortunately I don't think we have such a use case currently. If we do, I think we should tackle it as a separate effort since, like I said, we'd need more than just this PR to address it fully (the image service would also need two ports).

Reply:
When we use two ports, we just get a configuration of http port.
We do not have any insecure service URL, like we have The service URL.  We currently have the http port as configuration parameter, which we listen to.
How can we know what are the host and port of the insecure service URL? Maybe we should have another configuration parameter for this?  (InsecureServiceBaseURL) ?